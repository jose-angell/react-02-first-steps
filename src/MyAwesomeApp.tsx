import type { CSSProperties } from "react";

const firstName = 'Jose angel';
const lastName = 'Gallardo';
const favoriteGames = ['Metal Gear', 'Death Stranding', 'Half Life'];
const isActive = false;
const address ={
    street: 'Calle falsa',
    number: 123,
    country: 'USA'
}  
const myStyles: CSSProperties = {

    backgroundColor: '#fafafa',
    borderRadius: 20,
    padding: 10,
    marginTop: 30
}
export function MyAwesomeApp(){

    return (
        <div data-testid="div-app">
            <h1 data-testid="first-name-little">{firstName}</h1>
            <h3>{lastName}</h3>
            <p className="mi-clase-favorita">{favoriteGames.join(', ')}</p>
            <h1>{isActive ? 'Activo' : 'No Activo'}</h1>
            <p>{address.street}</p>
            <p style={myStyles}
            >{JSON.stringify(address)}</p>
        </div>
    )
}