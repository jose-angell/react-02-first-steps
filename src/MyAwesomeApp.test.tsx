import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import {render, screen} from '@testing-library/react'

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
        const {container} = render(<MyAwesomeApp/>)
        //screen.debug()
        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain("Jose angel")
        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toContain("Gallardo")
    })
    // test('should render firstName and lastName - screen', () => {
    //     render(<MyAwesomeApp/>)
    //     screen.debug()
    //     const h1 = screen.getByRole('heading', {
    //         level: 1
    //     });
    //     expect(h1.innerHTML).toContain("Jose angel")
    //     const h3 = screen.getByRole('heading', {
    //         level: 3
    //     });
    //     expect(h3.innerHTML).toContain("Gallardo")
    // })
    test('should match snapshot', () => {
        const {container} = render(<MyAwesomeApp/>)
        expect(container).toMatchSnapshot()
    })
// se asegura de que el componente no cambie
    test('should match snapshot', () => {
        render(<MyAwesomeApp/>)
        expect(screen.getByTestId('div-app')).toMatchSnapshot()
    })
})