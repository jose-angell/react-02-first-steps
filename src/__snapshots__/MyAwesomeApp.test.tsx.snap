// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-little"
    >
      Jose angel
    </h1>
    <h3>
      Gallardo
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Metal Gear, Death Stranding, Half Life
    </p>
    <h1>
      No Activo
    </h1>
    <p>
      Calle falsa
    </p>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"street":"Calle falsa","number":123,"country":"USA"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-little"
  >
    Jose angel
  </h1>
  <h3>
    Gallardo
  </h3>
  <p
    class="mi-clase-favorita"
  >
    Metal Gear, Death Stranding, Half Life
  </p>
  <h1>
    No Activo
  </h1>
  <p>
    Calle falsa
  </p>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"street":"Calle falsa","number":123,"country":"USA"}
  </p>
</div>
`;
